src/qs8-vprelu/scalar.c.in

```diff
@@ -0,0 +1,142 @@
+// 3. Neither the name of the copyright holder nor the names of its contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//  
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+// OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+// OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+// OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+// EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//  
+//  
+// SPDX-License-Identifier: BSD-3-Clause
+
+$assert DATATYPE in ["QS8", "QU8"]
+$assert BATCH_TILE >= 1
+
+#include <assert.h>
+
+#include "xnnpack/math.h"
+#include "xnnpack/vbinary.h"
+
+
+$XINT8_T = {"QS8": "int8_t", "QU8": "uint8_t"}[DATATYPE]
+$OUTPUT_MIN = {"QS8": -128, "QU8": 0}[DATATYPE]
+$OUTPUT_MAX = {"QS8": 127, "QU8": 255}[DATATYPE]
+void xnn_${DATATYPE.lower()}_vprelu_ukernel__scalar_u${BATCH_TILE}(
+    size_t batch,
+    const ${XINT8_T}* input_a,
+    const ${XINT8_T}* input_b,
+    ${XINT8_T}* output,
+    const union xnn_${DATATYPE.lower()}_vprelu_scalar_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(${XINT8_T}) == 0);
+  assert(input_a != NULL);
+  assert(input_b != NULL);
+  assert(output != NULL);
+
+  const int32_t input_zero_point = params->scalar.input_zero_point;
+  const int32_t slope_zero_point = params->scalar.slope_zero_point;
+  const int32_t output_zero_point = params->scalar.output_zero_point;
+  const float vpositive_multiplier = params->scalar.positive_multiplier;
+  const float vnegative_multiplier = params->scalar.negative_multiplier;                                
+  const float voutput_min_less_zero_point = (int32_t) params->scalar.output_min - (int32_t) params->scalar.output_zero_point;
+  const float voutput_max_less_zero_point = (int32_t) params->scalar.output_max - (int32_t) params->scalar.output_zero_point;
+  const float vmagic_bias = 12582912.0f;
+  const int32_t vmagic_bias_less_output_zero_point = INT32_C(0x4B400000) - (int32_t) params->scalar.output_zero_point;
+
+  $if BATCH_TILE == 1:
+    do {
+      const int32_t va = (int32_t) *input_a++ - input_zero_point;
+      const int32_t vb = (int32_t) *input_b++ - slope_zero_point;
+      int32_t vacc = XNN_UNPREDICTABLE(va < 0) ? va * vb : va;
+      float vscale = XNN_UNPREDICTABLE(va >= 0) ? vpositive_multiplier : vnegative_multiplier;
```

Comment:
I think these conditions (`va < 0` vs. `va >= 0`) should be the same (reorder true/false cases to match).

(Several other instances)